session_ii/javascript-fundamentals/session1/homework/raluca/classes.js

```diff
@@ -0,0 +1,75 @@
+// 1. write a base class called RegularShape whose constructor receives the number
+//    of sides and the length of a side. write at least two derived classes that extend
+//    RegularShape, representing two different regular shapes,
+//    containing a method called getArea, which must return the area of the
+//    shape, using the two arguments that the parent class should receive.
+
+// 1. Write a base class called RegularShape whose constructor receives the number of sides and the length of a side. 
+// Write at least two derived classes that extend RegularShape, representing two different regular shapes, containing a method called getArea, which must return the area of the shape, using the two arguments that the parent class should receive.
+
+class RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		this.numberOfSides = numberOfSides;
+		this.sideLength = sideLength;
+		this.shapeName = shapeName;
+	}
+
+	calculateArea() {
+		const cot = 1 / Math.tan(Math.PI / this.numberOfSides)
```

Comment:
I had no idea there was a general formula that can be applied for all these regular shapes. :-D